legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/ClassMappingSecondPassBuilder.java

```diff
@@ -45,11 +46,15 @@ public SetImplementation visit(ClassMapping classMapping)
         if (classMapping.extendsClassMappingId != null)
         {
             String superSetId = classMapping.extendsClassMappingId;
-            SetImplementation superSet = HelperMappingBuilder.getAllClassMappings(parentMapping).detect(c -> c._id().equals(superSetId));
-            if (superSet == null)
+            ImmutableSet<SetImplementation> superSets = HelperMappingBuilder.getAllClassMappings(parentMapping).select(c -> c._id().equals(superSetId));
+            if (superSets.size() == 0)
             {
                 throw new EngineException("Can't find extends class mapping '" + superSetId + "' in mapping '" + HelperModelBuilder.getElementFullPath(parentMapping, this.context.pureModel.getExecutionSupport()) + "'", classMapping.sourceInformation, EngineErrorType.COMPILATION);
             }
+            if (superSets.size() > 1)
+            {
+                throw new EngineException("Duplicated class mappings found with ID '" + superSetId + "' in mapping '" + HelperModelBuilder.getElementFullPath(parentMapping, this.context.pureModel.getExecutionSupport()) + "'", classMapping.sourceInformation, EngineErrorType.COMPILATION);
```

Comment:
Is there more details that we can provide to the user that will allow them to find the offending duplicates?  On big mappings that might import/extend other mappings, this can be a dauting task, hence we should provide as much trouble shoot data as possible.